Frontend/src/pages/registrations/components/RegistrationList.jsx

```diff
@@ -1,101 +1,112 @@
-import React, { useState } from 'react'
-import deleteRegistration from '../../../api/registration/delete/delete_registration'
-import getRegistrations from '../../../api/registration/get/get_registrations'
-import updateRegistration from '../../../api/registration/patch/update_registration'
+import { NonInteractiveTable } from '@thewca/wca-components'
+import React, { useEffect, useMemo, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import getCompetitionInfo from '../../../api/competition/get/get_competition_info'
+import { getConfirmedRegistrations } from '../../../api/registration/get/get_registrations'
+import getCompetitorInfo from '../../../api/user/get/get_user_info'
 import styles from './list.module.scss'
-import StatusDropdown from './StatusDropdown'
 
-function RegistrationRow({
-  competitorId,
-  eventIDs,
-  serverStatus,
-  competitionID,
-  setRegistrationList,
-  registrationList,
-}) {
-  const [status, setStatus] = useState(serverStatus)
+export default function RegistrationList() {
+  const { competition_id } = useParams()
+  const [isLoading, setIsLoading] = useState(true)
+  const [registrations, setRegistrations] = useState([])
+  const [heldEvents, setHeldEvents] = useState([])
```

Comment:
This `heldEvents` state plus corresponding effect to fetch happens in a lot of components. It might be worth creating a custom hook and calling that instead:
```
const { isLoading, heldEvents } = useHeldEvents(competition_id)
```
and the hook:
```
function useHeldEvents(competition_id) {
  const [isLoading, setIsLoading] = useState(true);
  const [heldEvents, setHeldEvents] = useState(null); // or [] instead of null, as you have it now

  useEffect(() => {
    getCompetitionInfo(competition_id).then((competitionInfo) => {
      setHeldEvents(competitionInfo.event_ids);
      setIsLoading(false);
    })
  }, [competition_id])

  return { isLoading, heldEvents };
}
```

Similarly for other things which are fetched more than 2 or 3 times.

Reply:
did not know about this, done!